### ISSUE TITLE
runner: Parses feature-files that are not provided

### ISSUE THREAD

**Issue by jenisys** [hrcorval/behavex#76]
**Describe the bug**
The runner tries to run feature-files that where not provided as command-line args (but only reside in the the same directory).

RELATED TO: #80

**To Reproduce**
0. Instrument `utils.should_feature_be_run()` by providing a print statement

```python
def should_feature_be_run(path_feature):
    print("XXX should_feature_be_run: %s" % path_feature)
    ... # ORIGINAL CODE is here.
```
2. Run `behave features/runner.*.feature` (from: behave repo)
3. Observe the lines with `XXX: ...` like: 
    GOOD_CASE: `XXX: .../features/runner.<SOMETHING>.feature` (OK)
    BAD_CASE: `XXX: .../features/step_dialect.generic_steps.feature` (not containing `features/runner.`)
    NOTE: All of the BAD_CASES where not passed as command-line args. 

**Expected behavior**
Try to only files/paths that were provided as command-line args.
Otherwise, you are wasting CPU unnecessarily.

**Version Info:**
 - behavex v2.0.2 (current HEAD of repository)

**Additional context**

**Comment by hrcorval** [comment]
Hi @jenisys 
First of all, I want to express my gratitude for taking the time to review this wrapper and report the defects you have identified. We are aware of the compatibility gaps with Behave when attempting to utilize the solution beyond basic usage. There is ample room for improvement, and your feedback is highly appreciated, especially considering your significant contribution to the community by developing the exceptional framework like Behave. In my opinion, Behave stands out as the finest multi-purpose BDD-based test automation framework written in Python. 
Having that said, We will carefully review all the reported issues and prioritize the corresponding fixes.

**Comment by anibalinn** [comment]
Hi @jenisys
First of all, I want to express my gratitude for taking the time to review this wrapper and report the defects you have identified. We are aware of the compatibility gaps with Behave when attempting to utilize the solution beyond basic usage. There is ample room for improvement, and your feedback is highly appreciated, especially considering your significant contribution to the community by developing the exceptional framework like Behave. In my opinion, Behave stands out as the finest multi-purpose BDD-based test automation framework written in Python.
Having that said, We will carefully review all the reported issues and prioritize the corresponding fixes.

**Comment by jenisys** [comment]
@anibalinn @hrcorval 
Thanks.

If you have time and want to explore it, the newest versions on the Github repo supports the `runner` as an extension point, meaning you can plug in you own `runner class`. You will be somewhat limited regarding the command-line options, but a `—jobs <NUMBER_OF_JOBS>` option is already provided for `parallel runners` that need it. The remaining additional command-line options would need to use `userdata defines`, like`behave -D my_runner.some_option=<VALUE> …` .

SEE ALSO:

* [features/runner.use_runner_class.feature](https://github.com/behave/behave/blob/main/features/runner.use_runner_class.feature)

NOTES:
I really liked the HTML report, it is really awesome.
There is a [cucumber/tag-expressions](https://github.com/cucumber/tag-expressions/tree/main/javascript) library (JavaScript implementation exists) that is also part of the newest `behave` versions that would probably make the `select-by-tag` mechanism more flexible, like:

The user enters a tag-expression: `(@smoke or @wip) and not @xfail` (in a TextEdit field; in addition to select a tag from the list)

**Comment by anibalinn** [comment]
Fix released in version 3.0.0: https://pypi.org/project/behavex/3.0.0/
branch [release_3.0.0](https://github.com/hrcorval/behavex/tree/release_3.0.0)
Thanks!
